### PR TITLE
fix(primary names): fix undername parsing issues

### PR DESCRIPTION
**Pull request by atticusofsparta**
## Summary by CodeRabbit

- **Bug Fixes**
  - Improved name parsing to correctly extract undernames when names contain dashes or complex segments, avoiding false matches and returning nil when appropriate.

- **Tests**
  - Added unit and integration tests covering dash-containing and edge-case name formats.
  - Added a scenario verifying undername owners can approve primary names for themselves.
  - Fixed several test assertions for accurate expected/actual ordering.

### CODE REVIEW

File: src/common/utils.lua [hunk 1]
```diff
@@ -450,7 +450,9 @@ function utils.undernameForName(name)
 	end
 
 	local baseName = utils.baseNameForName(name)
-	return string.gsub(name:reverse(), baseName:reverse() .. "_", "", 1):reverse()
+	-- Escape the pattern to treat it as literal text, not a pattern
+	local pattern = string.gsub(baseName:reverse() .. "_", "([%-%.%+%[%]%(%)%$%^%%%?%*])", "%%%1")
+	return string.gsub(name:reverse(), pattern, "", 1):reverse()
```
Comment (dtfiedler) [inline src/common/utils.lua:455]: use the technique @arielmelendez added to ar-io-network-proesss in [d120437](https://github.com/ar-io/ar-io-network-process/pull/434/commits/d120437b14ae6ca072e324982c31ed862768aff5)

**Review by dtfiedler — CHANGES_REQUESTED** [review]
simplify regex

File: src/common/utils.lua (hunk 1 above)
Comment (atticusofsparta) [reply src/common/utils.lua:455]: https://github.com/ar-io/ar-io-ant-process/pull/131/commits/374b60d6aa936f6b2282a9f11408fb620f699bcb

**Review by dtfiedler — APPROVED** [review]
